ci/install_yq.sh

```diff
@@ -14,20 +14,32 @@ die() {
 	exit 1
 }
 
+function verify_yq_exists() {
+  local yq_path=$1
+  local yq_version=$2
+  local expected="yq (https://github.com/mikefarah/yq/) version $yq_version"
+  if [ -x  "${yq_path}" ] && [ "$($yq_path --version)"X == "$expected"X ]; then
+    exit 0
```

Comment:
I don't think we necessarily want to exit the shell here - we may still want to execute further code. I think we should return a bool here and check the value in the caller.

Reply:
By extracting this method I haven't changed previous behaviour -> it was always stopping further processing if main purpose of this script was met (installed yq in correct version). This PR only fix existing problem with enforcing sudo for non GO_PATH installation of yq

Reply:
Behavior might be the same with the current code base but future contributions might change that, e.g. the function can be reused by some other script with different assumptions. It is generally a bad idea for a function to `exit` unless it reaches some fatal error condition that *really* prevents further execution. It should rather stick to returning some status and let the caller decide about the outcome.

Reply:
ok - updated approach